internal/services/machinelearning/machine_learning_compute_instance_resource.go

```diff
@@ -215,10 +214,16 @@ func resourceComputeInstanceCreate(d *pluginsdk.ResourceData, meta interface{})
 		computeInstance.Properties.ComputeInstanceAuthorizationType = pointer.To(machinelearningcomputes.ComputeInstanceAuthorizationType(authType))
 	}
 
+	mlWorkspacesClient := meta.(*clients.Client).MachineLearning.Workspaces
+	workspace, err := mlWorkspacesClient.Get(ctx, *workspaceID)
+	if err != nil {
+		return err
+	}
+
 	parameters := machinelearningcomputes.ComputeResource{
```

Comment:
Can we add a comment here explaining why we are using the `workspace.Model.Location` here instead of the `location` from the configuration file?
```suggestion
	// NOTE: Tracked by internal MS Bug 25419114: ARM does not require
	// proxy resource to have location field and should not be using it.
	// But due to a bug in ARM, Location field is being used causing the
	// the identity to be orphaned because the call is being routed to the
	// wrong location.
	parameters := machinelearningcomputes.ComputeResource{
```